### PR TITLE
[StableHLO] Port Philox rng

### PR DESCRIPTION
**Pull request by jvstokes**
This ports the philox rng implementation from mhlo to StableHLO, along with tests.

Addresses Issue #11761

### CODE REVIEW

File: compiler/src/iree/compiler/InputConversion/StableHLO/StableHLOToLinalgRandom.cpp [hunk 1]
```diff
@@ -446,6 +451,259 @@ LogicalResult generateLinalgThreeFry64(OpBuilder &builder, Location loc,
...
+  // Split into the 2xi32 used for threefry.
+  std::pair<ArithOpBuilder, ArithOpBuilder> input = splitI64(index);
+  ArithOpBuilder input0 = input.first;
+  ArithOpBuilder input1 = input.second;
+
+  // We initialize the state as such to match the XLA implementation.
+  PhiloxState state4 = {input0, input1, key.first, key.second};
+
+  // We perform 10 rounds to match the XLA implementation.
+  static const int kNumRounds = 10;
```
Comment (kuhar) [inline compiler/src/iree/compiler/InputConversion/StableHLO/StableHLOToLinalgRandom.cpp]: nit: use constexpr

**Review by kuhar — APPROVED** [review]
LGTM

**Comment by kuhar** [comment]
(I retitled the PR to match the conventions used in the other StableHLO porting work)

File: compiler/src/iree/compiler/InputConversion/StableHLO/StableHLOToLinalgRandom.cpp [hunk 2]
```diff
@@ -490,6 +765,16 @@ struct RngBitGeneratorConverter final
       return success();
     }
 
+    if (op.getRngAlgorithm() == mlir::stablehlo::RngAlgorithm::PHILOX) {
```
Comment (rsuderman) [inline compiler/src/iree/compiler/InputConversion/StableHLO/StableHLOToLinalgRandom.cpp]: Include for `mlir::stablehlo::RngAlgorithm::DEFAULT` case

**Review by rsuderman — CHANGES_REQUESTED** [review]

**Comment by jvstokes** [comment]
Reopening (accidentally closed due to a bad sync.)

**Review by rsuderman — APPROVED** [review]

**Pull request merged**